packages/store-api/src/platforms/vtex/resolvers/validateCart.ts

```diff
@@ -0,0 +1,162 @@
+import type { IStoreCart, IStoreOffer } from '../../../__generated__/schema'
+import type {
+  OrderForm,
+  OrderFormItem,
+  OrderFormInputItem,
+} from '../clients/commerce/types/OrderForm'
+import type { Context } from '..'
+
+const getId = (item: IStoreOffer) =>
+  [item.itemOffered.sku, item.seller.identifier, item.price].join('::')
+
+const orderFormItemToOffer = (
+  item: OrderFormItem,
+  index?: number
+): IStoreOffer & { index?: number } => ({
+  listPrice: item.listPrice / 100,
+  price: item.sellingPrice / 100,
+  quantity: item.quantity,
+  seller: { identifier: item.seller },
+  itemOffered: {
+    sku: item.id,
+    image: [],
+    name: item.name,
+  },
+  index,
+})
+
+const offerToOrderItemInput = (
+  offer: IStoreOffer & { index?: number }
+): OrderFormInputItem => ({
+  quantity: offer.quantity,
+  seller: offer.seller.identifier,
+  id: offer.itemOffered.sku,
+  index: offer.index,
+})
+
+const groupById = (offers: IStoreOffer[]): Map<string, IStoreOffer> =>
+  offers.reduce((acc, item) => {
+    const id = getId(item)
+
+    acc.set(id, acc.get(id) ?? item)
+
+    return acc
+  }, new Map<string, IStoreOffer>())
+
+const equals = (of1: OrderForm, of2: OrderForm) => {
+  const pick = ({ orderFormId, messages, items, salesChannel }: OrderForm) => ({
+    orderFormId,
+    messages,
+    salesChannel,
+    items: items.map(
+      ({ uniqueId, quantity, seller, sellingPrice, availability }) => ({
+        uniqueId,
+        quantity,
+        seller,
+        sellingPrice,
+        availability,
+      })
+    ),
+  })
+
+  return JSON.stringify(pick(of1)) === JSON.stringify(pick(of2))
+}
+
+/**
+ * This resolver implements the optimistic cart behavior. The main idea in here
+ * is that we receive a cart from the UI (as query params) and we validate it with
+ * the commerce platform. If the cart is valid, we return null, if the cart is
```

Comment:
Not so sure about returning `null` to signify that "no changes were made", or "the cart is valid". 

Why not add a field to the returned `orderForm` like `updated`, which would be a boolean value? Then we always return back the `orderForm`, but with an easy way for the frontend to check if the cart should be updated or not

Reply:
Just to save data. These carts can be big and it would be nice to save a little bit of bandwidth when the cart is valid

Reply:
What do you think?

Reply:
Yeaah, makes sense! 

I don't know how users will react to this hahahaha they might not find it weird. But regardless, from a performance standpoint, I agree. This is a great way to save data when dealing with big carts 😄  Especially taking into account poor mobile connections.